### DISCUSSION TITLE
PWM Override for each motor individually

### DISCUSSION THREAD

**Discussion by Banobel** [bitcraze/discussions#479] [Q&A]
Hello Community,

i am searching for a way to set the motor pwm, whilst bypassing the sabilizer to get a constand pmw signal on each motor.

The main Idea is to use the Crazyflie as a ESC-Progamming device via a small python-script.

The whole Story:

I have a Crazyflie 2.1 and i am using a BigQuad deck. As stated in the 'Bolt and BQ Deck - Workshop' (https://www.youtube.com/watch?v=xiWLhr-HpG8&t=1227s) [Timestamp: 19:35] ESC's need a programming adapter. Upon verifing this statement, my ESC gave me the Error code for an inconsisten throttle signal.
So i was thinking since the Desktop-Crazyflie-Controll-Panel sends the PID-Thrust values, i have inconsistencies on each motor, because the signals are mangled in the stabilizer. If i now bypass the stabilizer and set my motor ratios as i need them, i could get the initial start-up for my ESC's done and the ECS's are configured within the system boundries. 
The ECS's i am using are the 'Reely Sky-series-6A'

I also found a Thread discussing a similar topic: (https://forum.bitcraze.io/viewtopic.php?t=4011)
This Post seems quite helpfull, but my understanding of the code is quite lacking

so therefore i am asking for a little explanation and a code could look like.

Thanks in advance, if theres anything else you need to know please reach out

Best Regards
Banobel

**Comment by tobbeanton** [comment]
There is a bypass functionality so you can set individual motor pwm values. It can be performed by setting the [motorPowerSet](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/params/#motorpowerset) parameter values.

I suggest starting with the [log & param step-by-step guide](https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/user-guides/sbs_connect_log_param/) to get some understanding. Then use the [basicLedparamSync.py](https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/basicLedparamSync.py) as a base to write your own script that manipulates the motor PWMs using motorPowerSet parameter group.
